frontend/src/AlertsManagement/History.js

```diff
@@ -25,54 +48,68 @@ class History extends Component {
-        >
-          <ListItemText id={alertId}
-            disableTypography
-            primary={<Typography variant="body1" >Alert on 
-              <Box fontWeight='fontWeightBold' display='inline' m={1} style={{ color: '#0FA3B1'}}>
-                {` ${alert.timestampDate} (${formatDate(alert.timestampDate)})`}
-              </Box> has
-              <Box display='inline' m={1} style={{ color: '#0FA3B1'}}>
-                {` ${alert.anomalies.length} anomalies`}
-              </Box> with status
-              <Box display='inline' m={1} style={{ color: '#0FA3B1'}}>
-                {` ${alert.status}`}
-              </Box>
-              </Typography>
-            }
-          />
-        </ListItem>
-      );
-    });
-    return alertItems;
-  }
-
   render() {
     const styles = ({
       root: {
         flexGrow: 1,
-        maxWidth: 900,
+        maxWidth: '90%',
         margin: 'auto',
       },
+      header: {
+        marginTop: '50px',
+        marginBottom: '30px'
+      },
+      chip: {
+        margin: '5px',
+      }
     });
+
     const { allAlerts } = this.state;
     
-    if (allAlerts === null) {
-      return <div />;
-    }
+    if (allAlerts === null) return <div />;
 
     return (
       <div style={styles.root}>
-        <Typography variant="h4" gutterBottom style={{ marginTop: '50px'}}>Alerts Archive</Typography>
-        <List className="all-alerts-list">
-          {this.createListItems(allAlerts)}
-        </List>
+        <Typography variant="h4" gutterBottom style={styles.header}>Alerts Archive</Typography>
+        
+        <MaterialTable icons={TABLE_ICONS}
+          actions={[
+            {
+              icon: () => <LinkIcon />,
+              tooltip: 'See More',
+              onClick: (_event, rowData) => this.props.history.push(`alerts/${rowData.id}`),
+            }
+          ]}
+          columns={[
+            { title: 'Timestamp', field: 'timestampDate', filtering: false },
+            { title: '# of Anomalies', field: 'numAnomalies' },
+            { title: 'Status', field: 'status', sorting: false },
+            { title: 'Priority', field: 'priority', type: 'numeric', align: 'left' },
+            { 
+              title: 'Metrics', field: 'metrics', sorting: false,
+              render: rowData => rowData.metrics.split(DATA_DELIMITER).map((metric) => 
+                <Chip label={metric} color="secondary" variant="outlined" 
+                  size="small" style={styles.chip}
+                />)
+            },
+            { 
+              title: 'Dimensions', field: 'dimensions', sorting: false,
+              render: rowData => rowData.dimensions.split(DATA_DELIMITER).map((dimension) => 
+                <Chip label={dimension} color="primary" variant="outlined" 
+                  size="small" style={styles.chip}
+                />)
+            }
+          ]}
+          data={rows(allAlerts)} 
```

Comment:
so the history component isn't doing any fetching of data right? it just loads all of the stuff that's passed in , but doesnt do any actual calls?

and in addition, user arent able to do any form of editing (except for frontend-based filtering/sorting) based on the *currently visualized/seen* alerts (but not other alerts that are in the backend but might not be displayed in this current page for whatever reason), is my understanding correct?

Reply:
Yeah they can't edit the alerts on this page. You can actually allow it to be edited and I don't think it would be too difficult to include it, but I also felt that because being able to edit the alert's properties is already available on the alert's page and the Alerts page, wouldn't need to have it on the table as well.